### ISSUE TITLE
Add Openrouter

### ISSUE THREAD

**Issue by PierrunoYT** [supermemoryai/opensearch-ai#12]
Add Openrouter

**Comment by MaheshtheDev** [comment]
@PierrunoYT , can you be more specific exactly what feature you are expecting?

**Comment by PierrunoYT** [comment]
Hey OpenRouter where you can use different Models in case the tool will be paid only.

**Comment by Dhravya** [comment]
Yeah! I agree we should add open router. Since we're using the AI sdk this should not be that hard.

**Comment by AkshayBandi027** [comment]
Hey @Dhravya, I can take this up!
